zhaquirks/owon/pc321.py

```diff
@@ -0,0 +1,708 @@
+"""QUIRK FOR OWON PC321 (non-Tuya version)."""
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import Basic, Identify
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.smartenergy import Metering
+
+from zhaquirks import Bus, LocalDataCluster
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.owon import OWON
+
+OWON_PC321 = "PC321"
+OWON_PC321_CLUSTER_ID = 0xFD32
+
+"""Owon PC321 Attributes"""
+
+# Consumption
+OWON_METERING_ENERGY_CONSUMPTION_PH_A_ATTR = 0x4000
+OWON_METERING_ENERGY_CONSUMPTION_PH_B_ATTR = 0x4001
+OWON_METERING_ENERGY_CONSUMPTION_PH_C_ATTR = 0x4002
+OWON_METERING_TOTAL_ENERGY_CONSUMPTION_ATTR = 0x0000
+# Active power
+OWON_METERING_ACTIVE_POWER_PH_A_ATTR = 0x2000
+OWON_METERING_ACTIVE_POWER_PH_B_ATTR = 0x2001
+OWON_METERING_ACTIVE_POWER_PH_C_ATTR = 0x2002
+OWON_METERING_TOTAL_ACTIVE_POWER_ATTR = 0x0400
+# Reactive power
+OWON_METERING_REACTIVE_POWER_PH_A_ATTR = 0x2100
+OWON_METERING_REACTIVE_POWER_PH_B_ATTR = 0x2101
+OWON_METERING_REACTIVE_POWER_PH_C_ATTR = 0x2102
+OWON_METERING_TOTAL_REACTIVE_POWER_ATTR = 0x2103
+# Voltage
+OWON_METERING_VOLTAGE_PH_A_ATTR = 0x3000
+OWON_METERING_VOLTAGE_PH_B_ATTR = 0x3001
+OWON_METERING_VOLTAGE_PH_C_ATTR = 0x3002
+# Active current
+OWON_METERING_ACTIVE_CURRENT_PH_A_ATTR = 0x3100
+OWON_METERING_ACTIVE_CURRENT_PH_B_ATTR = 0x3101
+OWON_METERING_ACTIVE_CURRENT_PH_C_ATTR = 0x3102
+OWON_METERING_TOTAL_ACTIVE_CURRENT_ATTR = 0x3103
+# Reactive energy consumption
+OWON_METERING_REACTIVE_ENERGY_CONSUMPTION_PH_A_ATTR = 0x4100
+OWON_METERING_REACTIVE_ENERGY_CONSUMPTION_PH_B_ATTR = 0x4101
+OWON_METERING_REACTIVE_ENERGY_CONSUMPTION_PH_C_ATTR = 0x4102
+OWON_METERING_TOTAL_REACTIVE_ENERGY_CONSUMPTION_ATTR = 0x4103
+# Frequency
+OWON_METERING_AC_FREQUENCY_ATTR = 0x5005
+# Leakage
+OWON_METERING_LEAKAGE_CURRENT_ATTR = 0x3104
+
+METERING_CONSTANT_ATTRIBUTES = {
+    Metering.AttributeDefs.unit_of_measure.id: 0,
+    Metering.AttributeDefs.multiplier.id: 1,
+    Metering.AttributeDefs.divisor.id: 1000,
+    Metering.AttributeDefs.summation_formatting.id: 251,
+    Metering.AttributeDefs.metering_device_type.id: 0,
+    Metering.AttributeDefs.status: 0,
+}
+
+ELECTRICAL_MEASUREMENT_CONSTANT_ATTRIBUTES = {
+    ElectricalMeasurement.AttributeDefs.ac_voltage_multiplier.id: 1,
+    ElectricalMeasurement.AttributeDefs.ac_voltage_divisor.id: 10,
+}
+
+
+class OwonMetering(CustomCluster, Metering):
+    """Owon non-standard Metering cluster attributes, to be mapped into bus for later use in another clusters."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
```

Comment:
`_current_state` doesn't seem to be used. If so, this can be removed completely.